src/main/java/org/springframework/data/hazelcast/repository/query/HazelcastPartTreeQuery.java

```diff
@@ -118,6 +119,11 @@ public Object execute(Object[] parameters) {
             return this.executeDeleteQuery(query, queryMethod);
         }
 
+        if (this.isExists) {
+            final long count = this.keyValueOperations.count(query, queryMethod.getEntityInformation().getJavaType());
```

Comment:
The main reason why `exists` exists is to be able to differentiate it from `count`. The `count` operation needs to find all items matching our query whereas `exists` needs to find at least one and then can return the result immediately.

So, you need find a better performing implementation, but the structure looks good